Autopilot/Path Manager/MainBatterySensor.c

```diff
@@ -6,40 +6,44 @@
  */
 #include "main.h"
 #include "../Common/Common.h"
-#include "voltageSensor.h"
+#include "MainBatterySensor.h"
 
 
-char percent = 0;
-int currentVoltage = 0;
+int percent = 0;
+int batteryVoltage = 0;
 
 void __attribute__((interrupt, no_auto_psv)) _ADC2Interrupt(void)
 {
 
-    currentVoltage = ADC2BUF0;
+    batteryVoltage = ADC2BUF0;
+        
     IFS1bits.AD2IF = 0;		// Clear the ADC Interrupt Flag
 
 
 }
 
-void initBatterySensor(){
+void initMainBatterySensor(){
     //AN12 is the pin to get the battery information
     TRISBbits.TRISB12 = 1;
-    initVoltageADC();
+    initMainBatteryADC();
 
 }
 
-//float timeRemaining(){
-//    float dP = (currentPercent - lastPercent)/vTimeInterval; //Rate of change (Percent/second)
-//    float time = currentPercent/dP;
-//    return time;
-//}
-
-int getBatteryLevel(){
-    percent = currentVoltage;
-    return percent;
+char getMainBatteryLevel(){
+    //return percent of 2 cell battery. 
+    //To calculate percent, the formula takes into account the voltage divider, bit size of a char, minimum 6V of a LiPo 2S, and max charge of 8.4V
+    percent = (((double)(batteryVoltage)/3*13/128)-36*CELLS)*100/(6*CELLS);
```

Comment:
We are not going to be changing the picpilot battery very often, we always use a 2 cell Lipo, so this calculation won't change much.